File: contest/LeetCode_Weekly_Contest_203/pig0726/Maximum_Number_of_Coins_You_Can_Get.cpp
class Solution {
public:
    int maxCoins(vector<int>& piles) {
        sort(piles.begin(), piles.end(), [](const int a, const int b) { return a > b; });
        int n = piles.size()/3;
        int ret = 0;
        for (int i = 1, j = 0; j < n; i += 2, ++j)
            ret += piles[i];
        return ret;
    }
};

File: contest/LeetCode_Weekly_Contest_203/Maximum_Number_of_Coins_You_Can_Get.java
class Solution {
    public int maxCoins(int[] piles) {
        int n = piles.length / 3;
        Arrays.sort(piles);
        int sum = 0;
        for (int i = 3 * n - 2; i >= n; i -= 2) {
            sum += piles[i];
        }
        return sum;
    }
}

